include/seqan3/alignment/matrix/alignment_trace_matrix.hpp

```diff
@@ -0,0 +1,333 @@
+//!\cond
+    requires matrix_concept<alignment_score_matrix<score_matrix_params_t...>> &&
+             std::is_integral_v<typename alignment_score_matrix<score_matrix_params_t...>::entry_type>
+//!\endcond
+struct alignment_trace_matrix<database_type, query_type, align_config_type, alignment_score_matrix<score_matrix_params_t...>>
+    : public alignment_score_matrix<score_matrix_params_t...>
+{
+    //!\brief The type of the score matrix.
+    using score_matrix_type = alignment_score_matrix<score_matrix_params_t...>;
+
+    //!\brief The type of an entry in the score matrix.
+    using score_type = typename score_matrix_type::entry_type;
+
+    //!\copydoc seqan3::detail::matrix_concept::entry_type
+    using entry_type = trace_directions;
+
+    /*!\name Constructors, destructor and assignment
+     * The copy-constructor, move-constructor, copy-assignment, move-assignment,
+     * and destructor are implicitly defined.
+     * \{
+     */
+     alignment_trace_matrix() = default;
+     alignment_trace_matrix(alignment_trace_matrix const &) = default;
+     alignment_trace_matrix(alignment_trace_matrix &&) = default;
+     alignment_trace_matrix & operator=(alignment_trace_matrix const &) = default;
+     alignment_trace_matrix & operator=(alignment_trace_matrix &&) = default;
+
+    /*!\brief Construct the trace matrix by using a score_matrix.
+     * \param database     The database sequence.
+     * \param query        The query sequence.
+     * \param config       The alignment config.
+     * \param score_matrix The score matrix.
+     */
+    alignment_trace_matrix(database_type database, query_type query, align_config_type config, score_matrix_type score_matrix)
+        : score_matrix_type(std::move(score_matrix)),
+          _database{std::forward<database_type>(database)},
+          _query{std::forward<query_type>(query)},
+          _config{std::forward<align_config_type>(config)}
+    {}
+    //!\}
+
+    //!\copydoc seqan3::detail::matrix_concept::rows
+    using score_matrix_type::rows;
+    //!\copydoc seqan3::detail::matrix_concept::cols
+    using score_matrix_type::cols;
+
+    //!\brief The trace directions of the matrix at position (*row*, *col*).
+    inline entry_type at(unsigned row, unsigned col) const noexcept
+    {
+        entry_type direction{};
+
+        if(is_trace_diagonal(row, col))
+            direction |= entry_type::diagonal;
+
+        if(is_trace_up(row, col))
+            direction |= entry_type::up;
+
+        if(is_trace_left(row, col))
+            direction |= entry_type::left;
+
+        return direction;
+    }
+
+    //!\brief Access to the score_matrix.
+    inline score_matrix_type const & score_matrix() const noexcept
+    {
+        return *this;
+    }
+
+private:
+
+    //!\brief Does the trace come from the above entry?
+    inline bool is_trace_up(unsigned row, unsigned col) const noexcept
```

Comment:
parameters should be const qualified and I think `size_t` would be more appropriate than `unsigned`? We always use fixed-width integer types or `size_t` in SeqAn...